pageserver/src/tenant/layer_map.rs

```diff
@@ -269,6 +276,14 @@ where
         BatchedUpdates { layer_map: self }
     }
 
+    ///
+    /// Check if layer map contains layers with specified key and LSN range
+    ///
+    pub fn contains(&self, layer: Arc<L>) -> bool {
+        self.historic
+            .contains(historic_layer_coverage::LayerKey::from(&*layer))
```

Comment:
No need to consume the `Arc<L>`; `&Arc<L>` would work great. Compare against any `contains` methods in std: https://doc.rust-lang.org/stable/std/index.html?search=contains

Reply:
Sorry, I do not understand what do you suggest - replace Arc with &Arc?
It doesn't work because of Rust upcasting rules:
```
error[E0308]: mismatched types
    --> pageserver/src/tenant/timeline.rs:2865:33
     |
2865 |             if updates.contains(&l) {
     |                        -------- ^^ expected trait object `dyn PersistentLayer`, found struct `ImageLayer`
     |                        |
     |                        arguments to this function are incorrect
     |
     = note: expected reference `&Arc<(dyn PersistentLayer + 'static)>`
                found reference `&Arc<ImageLayer>`
```

Reply:
I saw your response in email but not on the site, wonder if you removed it?

Anyway it was about upcasting, I'll try and see.

Reply:
Aah it got removed oki. Resolving, feel free to unresolve if I've misunderstood.